### PR TITLE
Ypp syncing integration

### LINKED ISSUE #3589: Integration for leaving the program and turning off Ypp syncing
- [ ] turn off syncing from the settings tab
- [ ] leave the program

### PR DESCRIPTION
**Pull request by drillprop**
Fix #3589

### CODE REVIEW

File: packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx [hunk 1]
```diff
@@ -126,12 +236,7 @@ export const YppDashboardSettingsTab = () => {
             label="Category of imported videos"
             description="Choose a category to be assigned to the imported videos by default. You can change it for each video later once it’s imported."
           >
-            <Select
-              disabled={currentChannel?.shouldBeIngested}
-              items={categoriesSelectItems}
-              onChange={setCategory}
-              value={category}
-            />
+            <Select items={categoriesSelectItems} onChange={setCategoryId} value={categoryId} />
```
Comment (WRadoslaw) [inline packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx]: Shouldn't we disable the Select when the category is already chosen?

https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=3542%3A89393&t=YX2PO34nrAi7tXy2-0

Comment (drillprop) [reply packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx]: We changed that, and you should be able to update the category.

File: packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx [hunk 2]
```diff
@@ -29,31 +46,54 @@ const categoriesSelectItems: SelectItem[] =
...
-        `${atlasConfig.features.ypp.youtubeSyncApiUrl}/channels/${currentChannel.joystreamChannelId}`,
-        {
-          category,
-          isSync,
-        }
+        `${atlasConfig.features.ypp.youtubeSyncApiUrl}/channels/${currentChannel.joystreamChannelId}/ingest`,
+        { message, signature }
       )
 
       if (data.status === 200) {
```
Comment (WRadoslaw) [inline packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx:110]: Looks like refetch is not working as intended. After changing the category footer is still visible.

image

File: packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx [hunk 3]
```diff
@@ -156,7 +261,7 @@ export const YppDashboardSettingsTab = () => {
         secondaryButton={{
           text: 'Cancel',
           onClick: () => {
-            setCategory(currentChannel?.videoCategoryId)
+            setCategoryId(currentChannel?.videoCategoryId)
```
Comment (WRadoslaw) [inline packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx]:
```suggestion
            setCategoryId(currentChannel?.videoCategoryId)
            setIsSync(currentChannel?.shouldBeIngested)
```

File: packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx (hunk 2 above)
Comment (WRadoslaw) [reply packages/atlas/src/views/studio/YppDashboard/tabs/YppDashboardSettingsTab.tsx:110]: Same applies for `isSync`

**Review by WRadoslaw — CHANGES_REQUESTED** [review]

**Review by WRadoslaw — APPROVED** [review]
Works as expected!

Btw, WDYT about introducing some loading state to the settings tab?
RN, on entry the form is empty and after request is done it gets filled, which doesn't look good. If so let's create an issue. @toiletgranny @drillprop 

image

image

**Comment by toiletgranny** [comment]
Thanks for flagging, did not account for the settings taking that much time to load. Assuming we can not fetch the settings before the user goes specifically into the "Settings" tab, my quick recommendation are as follows:

1. If the loading is expected to take less than a second on average, I'd recommend a simple solution: reducing the opacity of the settings form to something like `.5` or even just `.75` and disabling any cursor events within that area.
2. If it's more likely to take more than a second, then I'd recommend using the Loader component (the largest variant or the slightly smaller one) and displaying that instead of the form's settings.

**Pull request merged**